source/MainPanel.cpp

```diff
@@ -132,10 +133,18 @@ void MainPanel::Step()
 	// and updating the isActive flag).
 	StepEvents(isActive);
 	
+	// Run the game, or pause the game
 	if(isActive)
-		engine.Go();
+	{	 
+		Audio::Resume();
```

Comment:
Should Resume and Pause be called every step? or only when the `isActive` state toggles from true to false or vice versa?

Reply:
Yes, if `isActive` were a toggle. Unfortunately it was defined on the fly. I made a new version where `isActive` lives a private life modified by Pause and Resume procedures. This gives you the behavior you wanted. Let me know if this is what you were thinking of.

Reply:
I was thinking more of defining another  local variable that has the original value of `isActive`, e.g. `wasActive`, before going through the steps that could modify isActive. Then, if no longer active, we call audio::pause.

I think there is not a similarly easy way to determine the exact step in which MainPanel becomes the top panel without a new class member variable, so we either call resume every active step (which should be low cost as the paused sources list will be empty except on the first active frame), or use the new class member with the local `isActive` to determine which frame is the first active one.

Reply:
Except then `isActive` and `wasActive` are reset on each step. There would be no tracking of `wasActive` across steps. Some variable needs to live outside the scope of `Step` in order to track what was active. That's my `paused` bool. Instead of comparing the two at the end I'm using switches to change the state. This allows you to compare and switch from anywhere.